spring-ai-core/src/main/java/org/springframework/ai/chat/history/ChatExchange.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.chat.history;
+
+import java.time.Instant;
+import java.util.List;
+
+import org.springframework.ai.chat.messages.Message;
+
+/**
+ *
+ * Represents single call request and response set of messages.
+ *
+ * @author Christian Tzolov
+ */
+public class ChatExchange {
```

Comment:
why not a java record ? (there's a number of javabeans/pojos in this PR that could be simplified as records)